calcculator-app/src/components/Input.js

```diff
@@ -0,0 +1,277 @@
+            secondDisplay:
+              prevState.input
+                .split(",")
+                .filter((el) => el !== ",")
+                .join("") + event,
+          }));
+        } else {
+          this.setState((prevState) => ({
+            input: prevState.input + event,
+            firstDisplay:
+              prevState.input
+                .split(",")
+                .filter((el) => el !== ",")
+                .join("") + event,
+          }));
+        }
+      }
+    } else {
+      if (this.state.mod) {
+        this.setState((prevState) => ({
+          input: parseInt([prevState.input, ...[event]].join("")),
+          secondDisplay: parseInt([prevState.input, ...[event]].join("")),
+        }));
+      } else {
+        this.setState((prevState) => ({
+          input: parseInt([prevState.input, ...[event]].join("")),
+          firstDisplay: parseInt([prevState.input, ...[event]].join("")),
+        }));
+      }
+    }
+  };
+
+  handleMod = (event) => {
+    if (event === "plusMinus") {
+      this.setState((prevState) => ({
+        input: Number(prevState.input) * -1,
+        firstDisplay: Number(prevState.input) * -1,
+        mod: event,
+        previous: prevState, // no real functionality just wanted to call prevState so it wouldnt stay faded
+      }));
+      console.log(this.state);
+    } else {
+      this.setState((prevState) => ({
+        input: "",
+        // firstDisplay: [...[answer]].join(""),
+        mod: event,
+        previous: prevState, // no real functionality just wanted to call prevState so it wouldnt stay faded
+      }));
+    }
+  };
+
+  handleClear = (event) => {
+    this.setState((prevState) => ({
+      input: [0],
+      firstDisplay: [],
+      mod: "",
+      secondDisplay: [],
+      answer: 0,
+    }));
+  };
+
+  handleEqual = (event) => {
+    console.log(event);
+    if (event.mod === "sum") {
+      let answer = Number(event.first) + Number(event.second);
+      console.log(answer);
+      this.setState((prevState) => ({
+        //mimicked first conditional of handleGrid in order for state to make answer first display which resets state each time and allows for cumulative functionality
+        input: answer.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ","),
+        firstDisplay: [...[answer]].join(""),
+        previous: prevState,
+        mod: "",
+      }));
+    }
+    if (event.mod === "subtract") {
+      let answer = Number(event.first) - Number(event.second);
+      this.setState((prevState) => ({
+        input: answer.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ","),
+        firstDisplay: [...[answer]].join(""),
+        previous: prevState,
+        mod: "",
+      }));
+    }
+    if (event.mod === "divide") {
+      let answer = Number(event.first) / Number(event.second);
+      this.setState((prevState) => ({
+        input: answer.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ","),
+        firstDisplay: [...[answer]].join(""),
```

Comment:
If we are going to repeat the use of the our regex on line 217 we should store it in a variable to keep our code DRYer.